packages/eslint-plugin/docs/rules/dot-notation.md

```diff
@@ -0,0 +1,40 @@
+# enforce dot notation whenever possible (`dot-notation`)
+
+## Rule Details
+
+This rule extends the base [`eslint/dot-notation`](https://eslint.org/docs/rules/dot-notation) rule.
+It supports all options and features of the base rule.
+
+## How to use
+
+```cjson
+{
+  // note you must disable the base rule as it can report incorrect errors
+  "dot-notation": "off",
+  "@typescript-eslint/dot-notation": ["error"]
+}
+```
+
+## Options
+
+New options
+
+- `allowPrivateClassPropertyAccess`
```

Comment:
Based on some recent standardisation of extension rule docs

````suggestion
See [`eslint/dot-notation`](https://eslint.org/docs/rules/dot-notation#options) options.
This rule adds the following options:

```ts
interface Options extends BaseDotNotationOptions {
  allowPrivateClassPropertyAccess?: boolean;
}

const defaultOptions: Options = {
  ...baseDotNotationDefaultOptions,
  allowPrivateClassPropertyAccess: false,
};
```

### `allowPrivateClassPropertyAccess`
````